### PR TITLE
Fixed some unit tests and add test step to run on Windows/Mac builds

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
Some of the unit tests had been failing on Windows, even though they were passing on CI (running Linux).
Because of this, it made sense to me to add a test step to the Windows and Mac builds as well, just to make sure we catch any weirdness between OSes.

I've also fixed what I could based on what failed locally for me.

```release-note
NONE
```

### CODE REVIEW

**Comment by devinbinnie** [comment]
@tboulis I was able to fix most of the failures, but couldn't figure out this one. I'm pretty sure there's something up with the regex we're using to detect Windows paths.

```
 FAIL  src/main/diagnostics/steps/internal/loggerHooks.test.js
  ● main/diagnostics/loggerHooks › should mask paths for all OSs › windows

    expect(received).toBe(expected) // Object.is equality

    Expected: 3
    Received: 2

       98 |             };
       99 |             const result = maskMessageDataHook(loggerMock)(message, 'file').data[0];
    > 100 |             expect(findOccurrencesInString(MASK_PATH, result)).toBe(3);
          |                                                                ^
      101 |             Object.defineProperty(process, 'platform', {
      102 |                 value: originalPlatform,
      103 |             });
```

**Review by m1lt0n — APPROVED** [review]

File: src/main/diagnostics/steps/internal/loggerHooks.test.js [hunk 1]
```diff
@@ -87,7 +94,7 @@ describe('main/diagnostics/loggerHooks', () => {
                 value: 'win32',
             });
             const message = {
-                data: ['C:/Users/user/Desktop/download.pdf C:/Users/user/Desktop/folder withSpace/file.txt'],
+                data: ['C:/Users/user/Desktop/download.pdf C:\\Users\\user\\Desktop\\folder withSpace/file.txt'],
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/main/diagnostics/steps/internal/loggerHooks.test.js:97]: It's not that, I added those to try and fix it and it was failing before. For some reason it's not showing 3 `MASK` constants, it's showing 2.

Also backslashes are how Windows shows its paths, so we should probably be testing for those.

[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/main/diagnostics/steps/internal/loggerHooks.test.js:97]: Feel free to add the changes to this branch, or I can cherry-pick them over. Whatever's easiest.

[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/main/diagnostics/steps/internal/loggerHooks.test.js:97]: Is there not 3?
`C:/Users/user/Desktop/download.pdf`
`C:\\Users\\user\\Desktop\\folder`
`withSpace/file.txt`

**Pull request merged**